source/features/conversation-timeline-filter.tsx

```diff
@@ -0,0 +1,174 @@
+	regenerateFilterSummary();
+	reapplySettings();
+
+	select.all('.' + menuItemCheckbox)
+		.forEach(element => {
+			element.style.display = 'none';
+		});
+
+	select(test)!.style.display = '';
+}
+
+function reapplySettings(): void {
+	for (const element of select.all('.js-timeline-item')) {
+		processTimelineItem(element);
+	}
+}
+
+// @ts-expect-error TODO: adjust to new select menu
+function restoreSettings(): void {
+	select.all<HTMLInputElement>(`#${showFilterName}`).forEach(element => {
+		element.checked = false;
+	});
+	select<HTMLInputElement>(`input[name=${showFilterName}][value="${currentSettings.valueOf()}"]`)!.checked = true;
+}
+
+function createRadio(title: string, summary: string, filterSettings: FilterSettings, checked: boolean): JSX.Element {
+	return (
+		<label className="select-menu-item d-flex" aria-checked={String(checked)} role="menuitemradio" tabIndex={0} onClick={async () => saveSettings(filterSettings, `#rgh-filter-menu-item-checkbox-${filterSettings}`)}>
+			<CheckIcon id={`rgh-filter-menu-item-checkbox-${filterSettings}`} style={{display: checked ? 'inherit' : 'none'}} className={`${menuItemCheckbox} select-menu-item-icon`} aria-hidden="true"/>
+			<div className="select-menu-item-text">
+				{title}
+				<div className="text-normal description">{summary}</div>
+			</div>
+		</label>
+	);
+}
+
+async function addTimelineItemsFilter(): Promise<void> {
+	select('#partial-users-participants')!.before(
+		<div className="discussion-sidebar-item js-discussion-sidebar-item">
+			<details className="details-reset details-overlay select-menu hx_rsm">
+				<summary className="text-bold discussion-sidebar-heading discussion-sidebar-toggle hx_rsm-trigger" aria-haspopup="menu" data-hotkey="x" role="button">
+					<GearIcon/>
+					<p>Filters</p>
+					<div id={timelineFiltersSelectorId}/>
+				</summary>
+
+				<details-menu className="select-menu-modal position-absolute right-0 hx_rsm-modal js-discussion-sidebar-menu" style={{zIndex: 99}}>
+					<div className="select-menu-header">
+						<span className="select-menu-title">Temporarily hide content</span>
+						<button className="hx_rsm-close-button btn-link close-button" type="button" data-toggle-for="reference-select-menu">
+							<XIcon aria-label="Close menu" role="img"/>
+						</button>
+					</div>
+					<div className="hx_rsm-content" role="menu">
+						{createRadio('Show all', '', FilterSettings.ShowAll, true)}
+						{createRadio('Show only comments', 'Hides commits and events', FilterSettings.ShowOnlyComments, false)}
+						{createRadio('Show only unresolved comments', 'Also hides resolved reviews and hidden comments', FilterSettings.ShowOnlyUnresolvedComments, false)}
+						{pageDetect.isPRConversation() && createRadio('Show only unresolved reviews', 'Also hides regular comments', FilterSettings.ShowOnlyUnresolvedReviews, false)}
+					</div>
+				</details-menu>
+			</details>
+
+			<p className="reason text-small text-gray"/>
```

Comment:
I don’t think this used anymore, there’s a div.description above, right?

Reply:
Done